hazelcast/src/main/java/com/hazelcast/cluster/ClusterState.java

```diff
@@ -65,13 +66,13 @@
  * @see NodeState
  * @since 3.6
  */
-public enum ClusterState {
+public enum ClusterState implements CodedEnum {
```

Comment:
This isn't legal. CodedEnum is part of our private API; ClusterState is part of public API. Now private API is leaking into public API.

Reply:
I was expecting something like it. Have to explore _uglier_ alternatives for those enums that are public.